src/robotFactory.js

```diff
@@ -1,17 +1,73 @@
 'use strict';
 
-function BaseRobot() {
-  // implement
+function BaseRobot(name, weight, coords, chipVersion) {
+  this.name = name;
+  this.weight = weight;
+  this.chipVersion = chipVersion;
+
+  this.coords = coords;
 }
 
-function FlyingRobot() {
-  // implement
+BaseRobot.prototype = {
+  ...BaseRobot.prototype,
+  goForward(step = 1) {
+    this.coords.y += step;
+  },
+
+  goRight(step = 1) {
+    this.coords.x += step;
+  },
+
+  goBack(step = 1) {
+    this.coords.y -= step;
+  },
+
+  goLeft(step = 1) {
+    this.coords.x -= step;
+  },
+
+  getInfo() {
+    return 'Robot: ' + this.name + ', Chip version: '
+    + this.chipVersion + ', Weight: ' + this.weight;
+  },
+};
+
+function FlyingRobot(name, weight, coords, chipVersion) {
+  BaseRobot.call(this, name, weight, coords, chipVersion);
+  this.coords.z = 0;
 }
 
-function DeliveryDrone() {
-  // implement
+FlyingRobot.prototype = {
+  ...BaseRobot.prototype,
+  goUp(step = 1) {
+    this.coords.z += step;
+  },
+  goDown(step = 1) {
+    this.coords.z -= step;
+  },
+};
+
+function DeliveryDrone(
+  name, weight, coords, chipVersion, maxLoadWeight, currentLoad) {
+  FlyingRobot.call(this, name, weight, coords,
+    chipVersion, maxLoadWeight, currentLoad);
```

Comment:
FlyingRobot нічого не працює з параметрами maxLoadWeight та currentLoad
```suggestion
  FlyingRobot.call(this, name, weight, coords,
    chipVersion);
```